integration/spark/shared/src/main/java/io/openlineage/spark/agent/util/PathUtils.java

```diff
@@ -75,14 +74,27 @@ public static DatasetIdentifier fromCatalogTable(
 
     Optional<URI> metastoreUri = extractMetastoreUri(sparkConf);
     // TODO: Is the call to "metastoreUri.get()" really needed?
-    //   Java's Optional should prevent the null in the first place.
+    // Java's Optional should prevent the null in the first place.
     if (metastoreUri.isPresent() && metastoreUri.get() != null) {
       // dealing with Hive tables
       DatasetIdentifier symlink = prepareHiveDatasetIdentifier(catalogTable, metastoreUri.get());
       return di.withSymlink(
           symlink.getName(), symlink.getNamespace(), DatasetIdentifier.SymlinkType.TABLE);
+    } else if (catalogTable.provider().isDefined()
```

Comment:
Minor: consider extracting this to separate method like `withAwsGlueSymlink` in case more of such conditions are added in future.

Reply:
This code generally calls for greater refactor :)